package.json

```diff
@@ -56,12 +57,17 @@
 		"autoprefixer": "check-npm-client && postcss -u autoprefixer -r --no-map --config packages/calypso-build/postcss.config.js",
 		"postcss": "check-npm-client && postcss -r --config packages/calypso-build/postcss.config.js",
 		"prebuild": "check-npm-client && yarn run install-if-deps-outdated",
-		"build": "check-npm-client && yarn run build-static && yarn run build-packages && yarn run build-css && run-p -s build-devdocs:* && run-p -s build-server build-client-if-prod build-client-if-desktop",
+		"build": "check-npm-client && yarn run build-static && yarn run build-packages && yarn run build-css && run-p -s build-devdocs:* && run-p -s build-server build-client-if-prod",
 		"build-css": "check-npm-client && run-p -s build-css:*",
 		"build-css:directly": "check-npm-client && node-sass --importer node_modules/node-sass-package-importer/dist/cli.js --include-path client client/assets/stylesheets/directly.scss public/directly.css --output-style compressed && yarn run -s postcss public/directly.css",
 		"build-css:editor": "check-npm-client && node-sass --importer node_modules/node-sass-package-importer/dist/cli.js --include-path client client/assets/stylesheets/editor.scss public/editor.css --output-style compressed && yarn run -s postcss public/editor.css",
 		"build-css:reader-mobile": "check-npm-client && node-sass --importer node_modules/node-sass-package-importer/dist/cli.js --include-path client client/assets/stylesheets/reader-mobile.scss public/reader-mobile.css --output-style compressed && yarn run -s postcss public/reader-mobile.css",
 		"build-devdocs:search-index": "check-npm-client && node bin/generate-devdocs-search-index.js",
+		"build-desktop": "check-npm-client && run-s build-desktop:config build-desktop:server build-desktop:client build-desktop:app",
+		"build-desktop:config": "check-npm-client && node desktop/bin/build-desktop-config.js",
+		"build-desktop:server": "check-npm-client && BROWSERSLIST_ENV=server CALYPSO_ENV=desktop webpack --config client/webpack.config.desktop.js --display errors-only",
+		"build-desktop:client": "check-npm-client && CALYPSO_ENV=desktop yarn run build-client-evergreen",
+		"build-desktop:app": "check-npm-client && cd desktop && electron-builder build",
```

Comment:
We'll want to ensure that additional arguments can be passed to the `build-desktop:app` script in order to override electron-builder's default behavior. One important optimization we make is building only the application executables if the desktop's config is unchanged (by passing the `"dir"` argument to electron builder). By adding this optimization, artifact build times in CI take ~5 minutes instead of ~18 minutes _after_ Calypso is built (see [PR](https://github.com/Automattic/wp-desktop/pull/869)).

Reply:
I _think_ something like `electron-builder ${ELECTRON_BUILDER_ARGS:-""} build --publish never` should work (i.e. we want a default, empty value of `ELECTRON_BUILDER_ARGS` that we can override as necessary).

Reply:
OK, I added the `ELECTRON_BUILDER_ARGS` option to the `build-desktop:app` task. Now I run the build like this:
```
NODE_ENV=production ELECTRON_BUILDER_ARGS=-c.mac.target=dir yarn run build-desktop
```
And the app build is indeed much faster! The `WordPress.com.app` directory is built just in 3 seconds, while the whole ZIP + DMG task takes 46 seconds.

Reply:
By the way, I noticed that the local build is incredibly slow if the `node_modules` directory contains unneeded packages. If any of them is a native one (i.e., has some `node-gyp` bindings that require C compilation), `electron-builder` rebuilds them for the target platform, and that's both slow and useless.

That's why the desktop build needs to happen inside a dedicated `desktop/` directory, and the `package.json` file there needs to be careful about what NPM modules it has in `dependencies`.

Reply:
> ... electron-builder rebuilds them for the target platform, and that's both slow and useless

This step may add some overhead but is necessary as native modules require compilation for their target platform.

Reply:
> That's why the desktop build needs to happen inside a dedicated `desktop/` directory, and the `package.json` file there needs to be careful about what NPM modules it has in `dependencies`.

The app doesn't need any native modules though. In our case, they are compiled only because `electron-builder` found them in the top-level `package.json` and thought that the app needs them. It didn't realize they are dev dependencies of the monorepo. The `electron-builder` doesn't recognize nuances like this.